src/hayhooks/server/utils/deploy_utils.py

```diff
@@ -218,13 +236,28 @@ def deploy_pipeline_files(app: FastAPI, pipeline_name: str, files: dict[str, str
 
     log.debug(f"Checking if pipeline '{pipeline_name}' already exists: {registry.get(pipeline_name)}")
     if registry.get(pipeline_name):
-        raise PipelineAlreadyExistsError(f"Pipeline '{pipeline_name}' already exists")
+        if not overwrite:
+            raise PipelineAlreadyExistsError(f"Pipeline '{pipeline_name}' already exists")
+        else:
+            log.debug(f"Clearing existing pipeline '{pipeline_name}'")
+            registry.remove(pipeline_name)
+
+            log.debug(f"Removing pipeline files for '{pipeline_name}'")
+            remove_pipeline_files(pipeline_name, settings.pipelines_dir)
```

Comment:
```suggestion
        log.debug(f"Clearing existing pipeline '{pipeline_name}'")
        registry.remove(pipeline_name)

        log.debug(f"Removing pipeline files for '{pipeline_name}'")
        remove_pipeline_files(pipeline_name, settings.pipelines_dir)
```

(maybe I'm wrong) could we remove `else`?

Reply:
I don't have a strong opinion :smile: